models/repo_indexer.go

```diff
@@ -360,3 +363,36 @@ func addOperationToQueue(op repoIndexerOperation) {
 		}()
 	}
 }
+
+// toUTF8DropErrors makes sure the return string is valid utf-8; attempts conversion if possible
+func toUTF8DropErrors(content []byte) []byte {
```

Comment:
Aren't there function for converting for templates already?

Reply:
Yes... almost.

* This function deals with invalid data differently: the templates function keeps invalid data around, while the new function removes all of it.

* I can't include `templates` from `models` because that would create a circular dependency. I could however move all the utf8 stuff from `templates` into `base` or something. That would put all byte/encoding stuff in one place, which makes sense. I was just trying to avoid messing too much with the existing code.

Reply:
If it is not too template specific it would be better to keep such code in base or utils

Reply:
OK. I'll work on it.

Reply:
I couldn't move it to `util` because of more dependencies. I added it to `base` in its own file.

Reply:
I couldn't move it to `util` because of more dependencies. I added it to `base` in its own file.